### ISSUE TITLE
Discussion: Join Wagtail Nest?

### ISSUE THREAD

**Issue by allcaps** [labd/wagtailstreamforms#226]
I'm not sure if this is something that would be worth discussing but would there be interest for this package to move to Wagtail Nest?

I'm on the Wagtail core team and actively involved in the Wagtail Nest initiative. Wagtail Stream Forms looks useful to the community and may need a bit more community support to keep updated.

Let me know if you would be interested. The Wagtail Nest documentation explains what this would mean. https://github.com/wagtail-nest

**Comment by Stormheg** [comment]
Would love to see this happen and happy to assist in any way I can!

**Comment by funkhaus-phil** [comment]
This would be such an amazing step for this project. It's always the same, that i can't upgrade my projects for at least half a year because i wanted to keep Streamforms... in that time wagtail just happens to be 2 main releases further :'(

**Comment by pimvernooij** [comment]
Yes this is something that will be picked up by us! Timelines tbd though! :-)

**Comment by stuartaccent** [comment]
I'm not involved anymore as have no need for it, but if you need anything from us shout. Good to see something I created to fill a whole for us still needed.

**Comment by roipoussiere** [comment]
> I'm not involved anymore as have no need for it, but if you need anything from us shout. Good to see something I created to fill a whole for us still needed.

Hello @stuartaccent,

First of all, thank you so much for all the work you have done for this project.

While it's totally understandable that you don't want to contribute on this project anymore since you don't use it, the fact is that some PRs are waiting for reviews and the project is not compatible with current Wagtail versions. It also depends on obsolete components like generic_chooser and wagtail_modeladmin.

As you mentioned, wagtailstreamforms is still used by some people who are waiting for some update.

Do you mind to accept @allcaps offer to transfer the project ownership to wagtail-nest, in order to allow other people to maintain this great project instead letting it die?

Thank you.

**Comment by Stormheg** [comment]
Hi @roipoussiere,

I've been in touch with some folks from Lab Digital during the Wagtail Space NL conference last week and there is interest in transferring to Wagtail Nest. Once that happens, would you be interested in helping out review open PRs and get the project back to good shape?

cc @marcelhekking @mvantellingen; hope we can make a transfer happen 🚀 🍀

**Comment by roipoussiere** [comment]
> Once that happens, would you be interested in helping out review open PRs and get the project back to good shape?

Yes I'll be glad to help.

**Comment by stuartaccent** [comment]
> Hello @stuartaccent,
> 
> First of all, thank you so much for all the work you have done for this project.
> 
> While it's totally understandable that you don't want to contribute on this project anymore since you don't use it, the fact is that some PRs are waiting for reviews and the project is not compatible with current Wagtail versions. It also depends on obsolete components like generic_chooser and wagtail_modeladmin.
> 
> As you mentioned, wagtailstreamforms is still used by some people who are waiting for some update.
> 
> Do you mind to accept @allcaps offer to transfer the project ownership to wagtail-nest, in order to allow other people to maintain this great project instead letting it die?
> 
> Thank you.

Hi @roipoussiere

A while ago I transferred the repo from us to labd as they offered to take ownership due to us not actively using it to keep it moving forward. So im not in a position to accept anything nor merge pr's thats Lab Digital's remit.

In terms of where it goes or its future im happy to help where I can.

Stu

**Comment by vinodMS** [comment]
Do you know if we have an update on this? are there any practical reasons preventing this move from happening?

**Comment by roipoussiere** [comment]
I have no idea.

But I worked a bit on this project and I can say that there is in fact a big technical dept to *get the project back to good shape*.

Wagtail has come a long way since. You could pick #225 to make it work with current Wagtail version, but it will depend on old components such as [wagtail_modeladmin](https://github.com/wagtail-nest/wagtail-modeladmin) which is now in maintenance mode.

So my solution was to start from the wagtail code base and just add a FormBuilder that uses a StreamField to define form fields, so I created [this PR](https://github.com/wagtail/wagtail/pull/12287) (a few hours ago, what a timing!). If it's not going to be merged, I will probably create a package.

**Comment by ethanaward** [comment]
We'd love for this to move forward, especially since the newest LTS version of Wagtail was just released and it would be great to have streamforms up to date sooner rather than later.

**Comment by roipoussiere** [comment]
Hello,

I just published [Wagtail Form Plugins](https://github.com/laas/wagtail-form-plugins), a project I was working on the past months. It's a set of plugins used to customize and improve the Wagtail form builder in a modular way. It adds streamfield-based forms, conditional fields, field templating, etc.

I hope other people will find this useful and will like the modular approach.

**Comment by vsalvino** [comment]
I've finally gotten around to getting [wagtail-flexible-forms](https://docs.coderedcorp.com/wagtail-flexible-forms/index.html) up to snuff.

Funny enough, it was created from a public kickstarter back in 2018 to be the official Wagtail stream forms module. I guess better late than never!
